uniffi_bindgen/src/bindings/python/templates/ErrorTemplate.py

```diff
@@ -0,0 +1,51 @@
+class RustError(ctypes.Structure):
+    _fields_ = [
+        ("code", ctypes.c_int32),
+        ("message", ctypes.c_char_p),
+    ]
+
+    def __str__(self):
+        return "RustError(code={}, message={})".format(self.code, str(self.message, "utf-8")) 
+
+{% for e in ci.iter_error_definitions() %}
+class {{e.name()}}:
+    {%- for value in e.values() %}
+    class {{value}}(Exception):
+        pass
+    {%- endfor %}
+
+    @staticmethod
+    def raise_err(err):
```

Comment:
If you don't want to use a finally block here, you could also try defining a `__del__` method on the `RustError` class, that calls the `string_free` FFI function on `self.message`. Either way, we do need to avoid leaking memory here.